subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/serialization/codecs/FileCollectionCodec.kt

```diff
@@ -53,33 +54,23 @@ class FileCollectionCodec(
     override suspend fun ReadContext.decode(): FileCollectionInternal {
         val contents = read()
         return if (contents is Collection<*>) {
-            fileCollectionFactory.create(UnpackingFileCollection(contents))
+            val configurableFiles = fileCollectionFactory.configurableFiles()
+            contents.forEach { element ->
+                when (element) {
+                    is File -> configurableFiles.from(element)
+                    is TransformationNode -> configurableFiles.from({ element.transformedSubject.get().files })
+                    is FileTree -> configurableFiles.from(element)
+                    else -> throw IllegalArgumentException("Unexpected item $element in file collection contents")
+                }
+            }
+            configurableFiles as FileCollectionInternal
```

Comment:
IIUC before the change the returned collection wasn't mutable, and now it is. Is this a desired change?

Reply:
The returned collection is now immutable.